python/lsst/sims/catUtils/utils/alertGenerator.py

```diff
@@ -0,0 +1,1475 @@
+                                                                       expmjd_list,
+                                                                       photometry_catalog,
+                                                                       dmag_cutoff)
+
+                q_f_dict = {}
+                q_m_dict = {}
+                q_snr_dict = {}
+
+                q_m_dict[0] = photometry_catalog.column_by_name('quiescent_lsst_u')
+                q_m_dict[1] = photometry_catalog.column_by_name('quiescent_lsst_g')
+                q_m_dict[2] = photometry_catalog.column_by_name('quiescent_lsst_r')
+                q_m_dict[3] = photometry_catalog.column_by_name('quiescent_lsst_i')
+                q_m_dict[4] = photometry_catalog.column_by_name('quiescent_lsst_z')
+                q_m_dict[5] = photometry_catalog.column_by_name('quiescent_lsst_y')
+
+                q_f_dict[0] = dummy_sed.fluxFromMag(q_m_dict[0])
+                q_f_dict[1] = dummy_sed.fluxFromMag(q_m_dict[1])
+                q_f_dict[2] = dummy_sed.fluxFromMag(q_m_dict[2])
+                q_f_dict[3] = dummy_sed.fluxFromMag(q_m_dict[3])
+                q_f_dict[4] = dummy_sed.fluxFromMag(q_m_dict[4])
+                q_f_dict[5] = dummy_sed.fluxFromMag(q_m_dict[5])
+
+                q_pmra = 1000.0*arcsecFromRadians(photometry_catalog.column_by_name('properMotionRa'))
+                q_pmdec = 1000.0*arcsecFromRadians(photometry_catalog.column_by_name('properMotionDec'))
+                q_parallax = 1000.0*arcsecFromRadians(photometry_catalog.column_by_name('parallax'))
+                q_ra = np.degrees(photometry_catalog.column_by_name('raICRS'))
+                q_dec = np.degrees(photometry_catalog.column_by_name('decICRS'))
+                q_tai = photometry_catalog.obs_metadata.mjd.TAI
+
+                q_snr_dict = {}
+                for i_filter in range(6):
+
+                    snr_template, local_gamma = calcSNR_m5(q_m_dict[i_filter],
+                                                           self.bp_dict[mag_names[i_filter]],
+                                                           obs_mag_cutoff[i_filter],
+                                                           photParams, gamma=gamma_template[i_filter])
+                    q_snr_dict[i_filter] = snr_template
+                    gamma_template[i_filter] = local_gamma
+
+                unq = photometry_catalog.column_by_name('uniqueId')
+
+                try:
+                    assert dmag_arr_transpose.shape == (len(chunk), len(mag_names), len(expmjd_list))
+                except AssertionError:
+                    print('dmag_arr_transpose_shape %s' % str(dmag_arr_transpose.shape))
+                    print('should be (%d, %d, %d)' % (len(chunk), len(mag_names), len(expmjd_list)))
+                    raise
+
+                # only include those sources for which np.abs(delta_mag) >= dmag_cutoff
```

Comment:
I left it out because, in a world where I wanted to swap `_filter_on_chip_name_then_photometry` back in, this would still go after that method.

The difference between this `dmag>dmag_cutoff` test and the one in `_filter_on_photometry_then_chip_name` is that, thanks to the `chipName` test, we now know at which time steps each object is actually detected (before, we assumed that all objects were observed at all time steps, regadless of whether or not they actually fell on a chip).

I am adding a comment to the effect of that last paragraph.

I think I would like to leave this code outside of whatever `_filter_on_x_then_y` method we use, since it is independent on the order of that initial filtering.

Feel free to tell me I am wrong.

Reply:
I think leaving it in with a comment explaining the above is fine. Without context it looks redundant, but it makes sense to leave it to maintain flexibility updating the code.